lib/spack/spack/cmd/arch.py

```diff
@@ -80,6 +92,7 @@ def display_target_group(header, target_group):
 
 def arch(parser, args):
     if args.generic_target:
+        tty.warn("spack arch --generic-target is deprecated in favor of --generic --target")
```

Comment:
Can we add when it will be removed?

Reply:
I'm also fine leaving it as a comment for now and add the warning in the next, and remove it in next next. It's a bit pedantic